src/engine/Account-Imap.cpp

```diff
@@ -191,134 +190,70 @@ buildTokenInfo(const char *key, const GValue *value, gpointer data)
     //  PINFO("buildTokenInfo: account '%s', token_count: '%" G_GINT64_FORMAT "'", (char*)key,
     //                  g_value_get_int64(value));
 
-    /* add the count to the total_count */
-    tokenInfo->total_count += g_value_get_int64(value);
+    /* add the count to the total_token_count */
+    tokenInfo->total_token_count += g_value_get_int64(value);
 
     /* allocate a new structure for this account and it's token count */
     this_account = (struct account_token_count*)
-                   g_new0(struct account_token_count, 1);
+                   new (struct account_token_count);
 
```

Comment:
In C++ the keywords "struct" and "class" designate a new type so it's not correct to include the keyword. Operator new always returns the correct pointer type so the cast is also unnecessary. In those places where you do need a cast, use the least-powerful C++ cast (usually static_cast unless you're casting away constness) that does the job.